### ISSUE TITLE
Spacetime (4.10.1): "error: assembler local symbol [...] not defined"

### ISSUE THREAD

**Issue by giltho** [ocaml/ocaml#10030]
Hi,

I'm not 100% sure this should be here, but there's a lot of chance it should.

# Error description

I'm getting the following error when building ocaml-lsp-server using ocaml 4.10.0+spacetime.
`error: assembler local symbol 'L189' not defined`
The server's implementation is in pure OCaml so I think the error comes from the ocaml compiler. The error seems to be coming from vendored merlin.

# Reproducing

## Process 
```sh
opam switch create 4.10.1+spacetime
eval $(opam env)
opam install ocaml-lsp-server
```

## My config
- OS: macOS Big Sur Version 11.0.1
- Hardware: MacBook Pro (16-inch 2019)

# More data

## Full error
```
    ocamlopt ocaml-lsp-server/vendor/merlin/src/kernel/.merlin_kernel.objs/native/mreader_explain.{cmx,o} (exit 2)
(cd _build/default && /Users/sacha/.opam/4.10.1+spacetime/bin/ocamlopt.opt -w -40 -w -a -g -I ocaml-lsp-server/vendor/merlin/src/kernel/.merlin_kernel.objs/byte -I ocaml-lsp-server/vendor/merlin/src/kernel/.merlin_kernel.objs/native -I ocaml-lsp-server/vendor/merlin/src/config/.config.objs/byte -I ocaml-lsp-server/vendor/merlin/src/config/.config.objs/native -I ocaml-lsp-server/vendor/merlin/src/dot-merlin/dot-protocol/.merlin_dot_protocol.objs/byte -I ocaml-lsp-server/vendor/merlin/src/dot-merlin/dot-protocol/.merlin_dot_protocol.objs/native -I ocaml-lsp-server/vendor/merlin/src/extend/.merlin_extend.objs/byte -I ocaml-lsp-server/vendor/merlin/src/extend/.merlin_extend.objs/native -I ocaml-lsp-server/vendor/merlin/src/ocaml/merlin_specific/.merlin_specific.objs/byte -I ocaml-lsp-server/vendor/merlin/src/ocaml/merlin_specific/.merlin_specific.objs/native -I ocaml-lsp-server/vendor/merlin/src/ocaml/parsing/.parsing.objs/byte -I ocaml-lsp-server/vendor/merlin/src/ocaml/parsing/.parsing.objs/native -I ocaml-lsp-server/vendor/merlin/src/ocaml/preprocess/.preprocess.objs/byte -I ocaml-lsp-server/vendor/merlin/src/ocaml/preprocess/.preprocess.objs/native -I ocaml-lsp-server/vendor/merlin/src/ocaml/typing/.typing.objs/byte -I ocaml-lsp-server/vendor/merlin/src/ocaml/typing/.typing.objs/native -I ocaml-lsp-server/vendor/merlin/src/ocaml/utils/.utils.objs/byte -I ocaml-lsp-server/vendor/merlin/src/ocaml/utils/.utils.objs/native -I ocaml-lsp-server/vendor/merlin/src/utils/.merlin_utils.objs/byte -I ocaml-lsp-server/vendor/merlin/src/utils/.merlin_utils.objs/native -intf-suffix .ml -no-alias-deps -o ocaml-lsp-server/vendor/merlin/src/kernel/.merlin_kernel.objs/native/mreader_explain.cmx -c -impl ocaml-lsp-server/vendor/merlin/src/kernel/mreader_explain.ml)
/var/folders/rp/lvrr9gq15rq7bts4ckdhfl5w0000gn/T/camlasmb3eb6f.s:3166:1: error: assembler local symbol 'L191' not defined

^
/var/folders/rp/lvrr9gq15rq7bts4ckdhfl5w0000gn/T/camlasmb3eb6f.s:3166:1: error: assembler local symbol 'L193' not defined

^
/var/folders/rp/lvrr9gq15rq7bts4ckdhfl5w0000gn/T/camlasmb3eb6f.s:3166:1: error: assembler local symbol 'L194' not defined

^
/var/folders/rp/lvrr9gq15rq7bts4ckdhfl5w0000gn/T/camlasmb3eb6f.s:3166:1: error: assembler local symbol 'L192' not defined

^
/var/folders/rp/lvrr9gq15rq7bts4ckdhfl5w0000gn/T/camlasmb3eb6f.s:3166:1: error: assembler local symbol 'L190' not defined

^
/var/folders/rp/lvrr9gq15rq7bts4ckdhfl5w0000gn/T/camlasmb3eb6f.s:3166:1: error: assembler local symbol 'L189' not defined

^
File "ocaml-lsp-server/vendor/merlin/src/kernel/mreader_explain.ml", line 1:
Error: Assembler error, input left in file /var/folders/rp/lvrr9gq15rq7bts4ckdhfl5w0000gn/T/camlasmb3eb6f.s
```

## Failing assembly input file

[camlasmb3eb6f.s](https://gist.github.com/giltho/b5cee70473afafdc13cd14c42032dc84)

**Comment by gretay-js** [comment]
Thank you for the detailed report.  It looks like the issue that was fixed in https://github.com/ocaml/ocaml/pull/9097 but the fix hasn't been backported to 4.10.  I am not sure what the best way to go about it. @gasche is it still possible to backport?

**Comment by xavierleroy** [comment]
The fix is part of 4.11, so I would suggest @giltho to use OCaml 4.11 instead of 4.10.

**Comment by giltho** [comment]
hmmm.. Some of my dependencies fail to build ocaml 4.11 so I haven't upgraded yet, but it's easier to PR those for the simple needed fixes rather than backport the spacetime fix into 4.10.

Thank you for your answer !!

**Comment by nojb** [comment]
> hmmm.. Some of my dependencies fail to build ocaml 4.11 so I haven't upgraded yet, but it's easier to PR those for the simple needed fixes rather than backport the spacetime fix into 4.10.
> 

Note also that Spacetime is no more in 4.12 -- it has been removed in favor of `statmemprof`.

**Comment by giltho** [comment]
Even more reasons to upgrade to 4.11 then !

**Comment by gasche** [comment]
> It looks like the issue that was fixed in #9097 but the fix hasn't been backported to 4.10. I am not sure what the best way to go about it. @gasche is it still possible to backport?

(Thanks for the question! I'm giving meta-level context in case it would be useful for another change next time.) I think that this would be the responsibility of @mshinwell or @stedolan, two maintainers who are more familiar with the work and have worked on it after the merge-decision was made.
